content/articles/Developing-simple-Blockchain-app-In-flutter/index.md

```diff
@@ -0,0 +1,291 @@
+
+It should be a private member variable so that other classes can't access it.
+
+```dart
+class Blockchain {
+    MqttClient _mqttClient = new MqttClient();
+
+AsyncTask<Void, Void, String>
+ asyncTask = new AsyncTask<Void, Void, String>() {
+
+ @Override  protected String doInBackground(Void... params) {
+
+return null;        }
+
+};
+```
+
+We can now run the `MqttClient` instance on a different thread. However, if we don't register it, we will not receive any messages. 
+
+Let's add an instance of the `MqttMessageHandler` class to our application and register our `_mqttClient` instance.
+
+```dart
+class Blockchain {
+
+    MqttClient _mqttClient = new MqttClient();
+
+    AsyncTask<Void, Void, String> asyncTask = new AsyncTask<Void, Void, String>() {
+
+            @Override  protected String doInBackground(Void... params) {
+
+    return null;         }
+
+    };
+
+MqttMessageHandler messageHandler = new MqttMessageHandler();
+
+_mqttClient.setHandler(messageHandler);
+```
+
+Now our `_mqttClient` instance will receive messages from the blockchain network. Finally, let's add some code to save the blockchain data to a file:
+
+```dart
+class Blockchain {
+
+    MqttClient _mqttClient = new MqttClient();
+
+AsyncTask<Void, Void, String> asyncTask = new AsyncTask<Void, Void, String>() {
+
+ @Override  protected String doInBackground(Void... params) {
+return null;         }
+@Override  protected void onPostExecute(String result) {
+getBlockHeight();
+Blockchain network = new Blockchain();
+   _mqttClient.connect("myDevices");
+```
+
+Now we can build our blockchain network by adding devices to it one at a time after they have registered with the same `MQTT` broker. 
+
+Fortunately, Flutter provides us with an easy way to do this. We only need to add a button and call the `addDevice()` function when it is clicked.
+
+```dart
+void main() {
+    runApp(new MyApp());
+}
+
+class MyApp extends StatelessWidget {
+    Widget build(BuildContext context) {
+        return new MaterialApp(
+        home: new Scaffold(
+        appBar: new AppBar(
+        title: new Text("My Blockchain"),
+        ),
+        body: new HomePage(), );
+            Button(
+                onClick: () {
+                network.addDevice();
+        });
+    } 
+}
+```
+
+Notice that we are connecting to `my Devices`, which is the name of my` MQTT` broker. You will need to change this to match the name of your broker. 
+
+When you run the application on different devices, they will all be connected to the same blockchain network and thus, share the blockchain data.
+
+### Conclusion
+Congratulations! You have now created a Flutter application that uses a Dart blockchain.
+
+This is just a simple blockchain application consisting of several devices. Each transaction is broadcasted to all the other devices
+
+You can access the full code from [this GitHub repository](https://github.com/johnniembugua/blockchain-app).
```

Comment:
The link you pasted redirects to a copy of this article. Push the complete code for this project to your Github repo. I wish to go through it.

Reply:
making changes to the repository adding the relevant files